src/NuGet.Services.Metadata.Catalog.Monitoring/Resources/PackageRegistrationMetadataResourceV2Feed.cs

```diff
@@ -39,9 +53,9 @@ public class PackageRegistrationMetadataResourceV2Feed : IPackageRegistrationMet
             return feedPackages.FirstOrDefault(p => p.Version == package.Version);
         }
 
-        private async Task<V2FeedPackageInfo> GetPackageFromLeaf(PackageIdentity package, ILogger log, CancellationToken token)
+        private Task<V2FeedPackageInfo> GetPackageFromLeaf(PackageIdentity package, ILogger log, CancellationToken token)
         {
-            return await _feedParser.GetPackage(package, log, token);
+            return _feedParser.GetPackage(package, log, token);
```

Comment:
Does GetPackageFromLeaf just return latest? Trying to understand how it's different from GetPackageFromIndex which does FirstOrDefault filter on version.